src/praisonai-agents/praisonaiagents/gateway/protocols.py

```
@@ -4683,6 +4687,166 @@ def evaluate(self, last_activity: float, now: float) -> LivenessDecision:
+    process (per event loop) exactly as the existing ``LockMap`` does, so
+    upgrading is byte-for-byte backward compatible. It provides **no** cross-
+    process guarantee — selecting a distributed backend (e.g. ``redis``) is what
+    extends serialisation across replicas.
+
+    The ``owner``/``ttl`` arguments are accepted for protocol symmetry but are
+    inert here: an in-process ``asyncio.Lock`` is released deterministically
+    when the holding task exits, so there is no crashed-holder lease to expire.
+    """
+
+    def __init__(self) -> None:
+        self._locks: Dict[str, "asyncio.Lock"] = {}
```

Ruff (0.16.0)

[error] Undefined name `asyncio`

(F821)